.github/workflows/provider.yml

```diff
@@ -0,0 +1,132 @@
+name: Build and release a provider
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        description: Provider name
+        required: true
+        type: string
+    secrets:
+      subject:
+        required: true
+        description: Capability provider issuer subject key
+      issuer:
+        required: true
+        description: Capability provider issuer key
+env:
+  REGISTRY: ghcr.io
+  REPOSITORY: ${{ github.repository }}
+jobs:
+  build:
+    strategy:
+      matrix:
+        arch: ["x86_64", "aarch64"]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: goto-bus-stop/setup-zig@v2
+
+      - name: Add musl targets
+        run: |
+          rustup target add ${{ matrix.arch }}-unknown-linux-musl
+
+      - name: Install cargo-zigbuild
+        run: |
+          cargo install cargo-zigbuild
+
+      - name: Build
+        run: |
+          cargo zigbuild --release --target ${{matrix.arch}}-unknown-linux-musl
+          name=$(echo ${{inputs.name}} | tr '-' '_')
+          mv target/${{matrix.arch}}-unknown-linux-musl/release/$name ${{inputs.name}}
+        working-directory: ${{inputs.name}}
+
+      - name: Store artifact
+        uses: actions/upload-artifact@v4
+        #if: startswith(github.ref, 'refs/tags/v') # Only run on tag push
+        with:
+          name: ${{inputs.name}}-${{matrix.arch}}
+          path: ${{inputs.name}}/${{inputs.name}}
+  release:
+    needs: [build]
+    #if: startsWith(github.ref, 'refs/tags/${{inputs.name}}-v')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Extract tag context
+        id: ctx
+        run: |
+          sha_short=$(git rev-parse --short HEAD)
+          echo "owner=${GITHUB_REPOSITORY_OWNER,,}" >> "$GITHUB_OUTPUT"
+          echo "sha_short=${sha_short}" >> "$GITHUB_OUTPUT"
+          echo "version=${GITHUB_REF#refs/tags/${{inputs.name}}v}" >> "$GITHUB_OUTPUT"
+          echo "artifact_name=$(echo ${{inputs.name}} | tr '- "')" >> "$GITHUB_OUTPUT"
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: wash-cli@0.27.0
```

Comment:
Now that wash-cli 0.28.0 is out, would it be worth pinning it to that? That would let you also get rid of the `crane` dependency since it outputs the `digest` as part of the `wash push` outputs